### PR TITLE
Implement fluent provenance query system

### PR DESCRIPTION
**Pull request by brianorwhatever**
Add a fluent provenance query system and convenience methods to Originals SDK to simplify filtering and searching migrations and transfers.

---
<a href="https://cursor.com/background-agent?bcId=bc-2d4740fe-0bc1-42ed-a7e7-f5c45c0d7918"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">Open in Cursor</picture></a>&nbsp;<a href="https://cursor.com/agents?id=bc-2d4740fe-0bc1-42ed-a7e7-f5c45c0d7918"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">Open in Web</picture></a>

## Summary by CodeRabbit

- New Features
  - Fluent provenance query API with date-range builders and chaining.
  - Filter migrations by source/destination layer, transaction ID, and inscription ID.
  - Filter transfers by from/to address and transaction ID.
  - Quick accessors: counts, first/last entries, and full results.
  - Provenance summary (created, creator, current layer, migration/transfer counts, last activity).
  - Convenience helpers for common migration/transfer queries.

- Tests
  - Comprehensive unit tests covering querying, chaining, date filters, performance, and edge cases.

### CODE REVIEW

File: src/lifecycle/ProvenanceQuery.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    const results = this.all();
+    return results.length > 0 ? results[results.length - 1] : null;
+  }
+
+  /**
+   * Get all results
+   */
+  all(): Array<Migration | Transfer> {
+    return [...this.provenance.migrations, ...this.provenance.transfers];
+  }
```
[the remark this answers is not in the document]
Comment (brianorwhatever) [reply src/lifecycle/ProvenanceQuery.ts:86]: @cursoragent In src/lifecycle/ProvenanceQuery.ts around lines 84 to 86, the all() method
currently returns the raw merged migrations and transfers and ignores
after()/before() filters and ordering; change all() to merge the two lists, sort
the combined list by timestamp (ascending), and then call
this.applyDateFilters(...) on that sorted array before returning so
afterDate/beforeDate are applied and first()/last()/count() operate on the
filtered, chronologically ordered result.

File: tests/unit/lifecycle/ProvenanceQuery.test.ts [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+      const result = largeAsset.queryProvenance()
+        .transfers()
+        .to('addr-500')
+        .after(new Date(Date.now() - 24 * 60 * 60 * 1000))
+        .all();
+      const endTime = Date.now();
+
+      expect(result.length).toBe(1);
+      expect(endTime - startTime).toBeLessThan(50);
+    });
```
[the remark this answers is not in the document]
Comment (brianorwhatever) [reply tests/unit/lifecycle/ProvenanceQuery.test.ts:537]: @cursoragent In tests/unit/lifecycle/ProvenanceQuery.test.ts around lines 507 to 537, remove
the hard real-time assertions that check endTime - startTime < 100 and < 50
because they make the test flaky; instead either delete those two expect timing
lines or replace them with deterministic checks (e.g., assert that the query
completed synchronously by checking no pending promises or using a mocked
timer/instrumentation to measure elapsed logical time) so the test only asserts
correctness of result.length and not wall-clock performance.
